skimage/measure/tests/test_colocalization.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import pytest
+from skimage.measure import (pearson_corr_coeff, manders_coloc_coeff,
+                              manders_overlap_coeff, intersection_coeff)
+
+
+def test_invalid_input():
+    # images are not same size
+    img1 = np.array([[i + j for j in range(4)] for i in range(4)])
+    img2 = np.ones((3, 5, 6))
+    mask = np.array([[i <= 1 for j in range(5)] for i in range(5)])
+    non_binary_mask = np.array([[2 for j in range(4)] for i in range(4)])
+
+    with pytest.raises(ValueError):
+        pearson_corr_coeff(img1, img1, mask)
+    with pytest.raises(ValueError):
+        pearson_corr_coeff(img1, img2)
+    with pytest.raises(ValueError):
+        pearson_corr_coeff(img1, img1, mask)
+    with pytest.raises(Warning):###
+        pearson_corr_coeff(img1, img1, non_binary_mask)
+    with pytest.raises(ValueError):
+        manders_coloc_coeff(img1, mask)
+    with pytest.raises(Warning):
+        manders_coloc_coeff(img1, non_binary_mask)
+    with pytest.raises(ValueError):
+        manders_coloc_coeff(img1, img1 > 0, mask)
+    with pytest.raises(Warning):
+        manders_coloc_coeff(img1, img1 > 0, non_binary_mask)
+    with pytest.raises(ValueError):
+        manders_overlap_coeff(img1, img1, mask)
+    with pytest.raises(ValueError):
+        manders_overlap_coeff(img1, img2)
+    with pytest.raises(ValueError):
+        manders_overlap_coeff(img1, img1, mask)
+    with pytest.raises(Warning):
+        manders_overlap_coeff(img1, img1, non_binary_mask)
+    with pytest.raises(ValueError):
+        intersection_coeff(img1 > 2, img2 > 1, mask)
+    with pytest.raises(Warning):
+        intersection_coeff(img1, img2)
+    with pytest.raises(ValueError):
+        intersection_coeff(img1 > 2, img1 > 1, mask)
+    with pytest.raises(Warning):
+        intersection_coeff(img1 > 2, img1 > 1, non_binary_mask)
+
+
+def test_pcc():
+    # simple example
+    img1 = np.array([[i + j for j in range(4)] for i in range(4)])
+    assert pearson_corr_coeff(img1, img1) == (1.0, 0.0)
+
+    img2 = np.where(img1 <= 2, 0, img1)
+    assert pearson_corr_coeff(img1, img2) \
+           == (0.944911182523068, 3.5667540654536515e-08)
+
+    # change background of roi and see if values are same
+    roi = np.where(img1 <= 2, 0, 1)
+    assert pearson_corr_coeff(img1, img1, roi=roi) \
+           == pearson_corr_coeff(img1, img2, roi=roi)
+
+
+def test_mcc():
+    img1 = np.array([[j for j in range(4)] for i in range(4)])
+    mask = np.array([[i <= 1 for j in range(4)]for i in range(4)])
+    assert manders_coloc_coeff(img1, mask) == 0.5
+
+
+def test_moc():
+    img1 = np.ones((4, 4))
+    img2 = 2 * np.ones((4, 4))
+    assert manders_overlap_coeff(img1, img2) == 1
```

Comment:
I see that these tests are all for integer dtypes. If images must be non-negative, we should add a test for that. If negative values are allowed, then please add additional tests for that case, possibly with floating point inputs.